### PR TITLE
 Use generic module system, add linux compatibility 

### PR DESCRIPTION
**Pull request by BlueDrink9**
I refactored out the OSX code a little more, and decentralised the checks on OS. I've tried to otherwise keep it true to the style you were going for, I think.

Let me know if there's anything to be changed.

### CODE REVIEW

File: tmux-wifi.tmux
Comment (gmoe) [inline tmux-wifi.tmux:4]: Sourcing `source_module.sh` here shouldn't be necessary since we're calling the files in `scripts/` below this. On macOS the plugin does not work for me without the line removed.

**Review by gmoe — CHANGES_REQUESTED** [review]
Looks fantastic to me aside from my one comment!

File: tmux-wifi.tmux
Comment (BlueDrink9) [reply tmux-wifi.tmux:4]: Sure, approve the edit and we're good to go! Might I suggest removing macos from your repo name? Old repos will be redirected to the old one without issues.

**Review by gmoe — APPROVED** [review]

File: tmux-wifi.tmux
Comment (gmoe) [reply tmux-wifi.tmux:4]: Sure thing, I approved the PR. I definitely plan on renaming the repo once everything is merged in!

Comment (BlueDrink9) [reply tmux-wifi.tmux:4]: You'll still need to merge it, btw. You're the only one that can do that...

**Comment by BlueDrink9** [comment]
If you're happy with it, is there a reason you haven't merged btw?

**Comment by gmoe** [comment]
@BlueDrink9 Haha sorry I was in work-mode there. In my brain you had write access so I was thinking _"I will let him merge it."_

I'll drop that source line and adjust the README/repo name, thanks for your help!

**Pull request merged**
